src/main/scala/scala/collection/next/NextIterableOnceOpsExtensions.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection
+package next
+
+import scala.language.implicitConversions
+
+private[next] final class NextIterableOnceOpsExtensions[A, CC[_], C](
+  private val col: IterableOnceOps[A, CC, C]
+) extends AnyVal {
+  import NextIterableOnceOpsExtensions.GroupMap
+
+  /**
+   * Partitions this IterableOnce into a map according to a discriminator function `key`. All the values that
+   * have the same discriminator are then transformed by the `value` function and then reduced into a
+   * single value with the `reduce` function.
+   *
+   * {{{
+   *   def occurrences[A](as: IterableOnce[A]): Map[A, Int] =
+   *     as.iterator.groupMapReduce(identity)(_ => 1)(_ + _)
+   * }}}
+   *
+   * @note This will force the evaluation of the Iterator.
+   */
+  def groupMapReduce[K, B](key: A => K)(f: A => B)(reduce: (B, B) => B): immutable.Map[K, B] =
+    groupMapTo(key)(f).reduce(reduce)
+
+  def groupByTo[K](key: A => K): GroupMap[A, K, A, immutable.Iterable, immutable.Map] =
+    groupMapTo(key)(identity)
+
+  def groupMapTo[K, V](key: A => K)(f: A => V): GroupMap[A, K, V, immutable.Iterable, immutable.Map] =
+    new GroupMap(col, key, f, immutable.Iterable, immutable.Map)
+}
+
+object NextIterableOnceOpsExtensions {
+  final case class GroupMap[A, K, V, CC[_], MC[_, _]](
+    col: IterableOnceOps[A, Any, Any],
+    key: A => K,
+    f: A => V,
+    colFactory: Factory[V, CC[V]],
```

Comment:
what's the rationale behind using a type `CC[_]` and not a type `C`?

Reply:
Actually none, I was just playing with the code.
I will try to use `C` since that should open the door to things like **BitSet**.